org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/ProgressReporterManager.java

```diff
@@ -274,6 +276,7 @@ protected void sendStatus() {
 					notification = endNotification;
 					progressId = UUID.randomUUID().toString();
 					sentBegin = false;
+					sentEnd = true;
 				} else {
 					var reportNotification = new WorkDoneProgressReport();
```

Comment:
In this area, you can do what we did a few lines above, `progressReport.setStatus(task + SEPARATOR + subTaskName);` . Except instead of `setStatus`, I think you want to call `setMessage` .